aten/src/ATen/native/mps/operations/Quantized.mm

```diff
@@ -601,6 +716,53 @@ kernel void kernel_mul_mv(
 
 )METAL_QUANTIZED");
 
+Tensor _convert_weight_to_int4pack_mps(const Tensor& in, int64_t innerKTiles) {
+  TORCH_CHECK(in.dim() == 2,
+      __func__, " : expect weight to be 2D tensor.");
+  TORCH_CHECK(in.dtype() == at::kInt,
+      __func__, " : expect weight to be kInt.");
+  TORCH_CHECK(innerKTiles == 2 || innerKTiles == 4 || innerKTiles == 8,
+      __func__, " : innerKTiles need to be 2, 4, or 8, got ", innerKTiles);
+
+  auto weight = in.contiguous();
+  auto N = weight.size(0);
+  auto K = weight.size(1);
+
+  // Create fake shapes for cpu. The meta registration in dynamo requires
+  // operator has the same output shape for each device. So creating a fake
+  // shape {N / 8, K / (16 * innerKTiles), 32, innerKTiles / 2}
+  auto weight_packed = at::empty(
+      {N / 8, K / (16 * innerKTiles), 32, innerKTiles / 2},
+      at::TensorOptions().dtype(at::kInt));
```

Comment:
I dont follow this one. This is using kInt dtype so each element is of 32bit. And looking at the code, I cant tell how many elements are supposed to be in this tensor. Is it ((N * K)/8)? I guess that kind of makes sense because 32bit int can pack 8 4bit values.